src/commands/annotatedTag.js

```diff
@@ -59,36 +59,27 @@ export async function annotatedTag ({
     }
 
     // Fill in missing arguments with default values
-    if (tagger === undefined) tagger = {}
-    if (tagger.name === undefined) {
-      tagger.name = await config({ fs, gitdir, path: 'user.name' })
-    }
-    if (tagger.email === undefined) {
-      tagger.email = await config({ fs, gitdir, path: 'user.email' })
-    }
+    tagger = { ...tagger }
+    tagger.name = tagger.name || await config({ fs, gitdir, path: 'user.name' })
+    tagger.email = tagger.email || await config({ fs, gitdir, path: 'user.email' })
     if (tagger.name === undefined || tagger.email === undefined) {
       throw new GitError(E.MissingAuthorError)
     }
 
-    const { type } = (await readObject({ fs, gitdir, oid }))
-    let taggerDateTime = tagger.date || new Date()
+    const taggerDateTime = tagger.date || new Date()
+    tagger.timestamp = tagger.timestamp != null
```

Comment:
This needs to be `tagger.timestamp !== undefined && tagger.timestamp !== null` unfortunately. (I've challenged Dan to think of the edge case... it's probably easier to figure out why this is needed for `timezoneOffset`

Reply:
Hmm, I'm totally not sure how loose equality works in this case, but I'll fix for now.

Reply:
(Apparently I also did not know how loose equality worked in this case. Dan schooled me!)